### PR TITLE
upgrade go-fil-markets, go-sectorbuilder, and go-storage-miner

### PR DESCRIPTION
**Pull request by ingar**
### Motivation
`go-fil-markets` has been updated to use the common types from `specs-actors`.

### Proposed changes
Updates the `go-fil-markets`, `go-storage-miner`, and `go-sectorbuilder` dependencies and migrates the implemented interfaces to use the new types.

### CODE REVIEW

**Comment by laser** [comment]
@anorth - Note: This PR needs to be merged before I can add the sector pre-commit expiry epoch parameter to the `SendPreCommitSector` method.

**Comment by laser** [comment]
@shannonwells - Note: This PR changes the types in `RetrievalClientNodeConnector` and `RetrievalProviderNodeConnector` slightly. It may be useful for you to branch off of this branch so that your retrieval connector-work doesn't conflict with this changeset.

File: internal/app/go-filecoin/storage_miner_connector/connector.go [hunk 1]
```diff
@@ -348,28 +366,29 @@ func (m *StorageMinerNodeConnector) GetSealTicket(ctx context.Context) (storage.
...
-func (m *StorageMinerNodeConnector) GetSealSeed(ctx context.Context, preCommitMsg cid.Cid, interval uint64) (seed <-chan storage.SealSeed, err <-chan error, invalidated <-chan struct{}, done <-chan struct{}) {
-	sc := make(chan storage.SealSeed)
-	ec := make(chan error)
-	ic := make(chan struct{})
-	dc := make(chan struct{})
+func (m *StorageMinerNodeConnector) GetSealSeed(ctx context.Context, preCommitMsg cid.Cid, interval uint64) (<-chan storagenode.SealSeed, <-chan storagenode.SeedInvalidated, <-chan storagenode.FinalityReached, <-chan storagenode.GetSealSeedError) {
+	sc := make(chan storagenode.SealSeed)
+	ec := make(chan storagenode.GetSealSeedError)
+	ic := make(chan storagenode.SeedInvalidated)
+	dc := make(chan storagenode.FinalityReached)
```
Comment (anorth) [inline internal/app/go-filecoin/storage_miner_connector/connector.go]: Could we manage less opaque names?

File: internal/app/go-filecoin/storage_market_connector/client.go [hunk 2]
```diff
@@ -161,41 +160,41 @@ func (s *StorageClientNodeConnector) ValidatePublishedDeal(ctx context.Context,
...
-		if err != nil {
-			return 0, err
+	for _, proposal := range msgProposals {
+		if proposal.Proposal == deal.Proposal {
+			var ret market.PublishStorageDealsReturn
+			err := encoding.Decode(chnMsg.Receipt.ReturnValue, &ret)
+			if err != nil {
+				return 0, err
+			}
+			sectorID := ret.IDs[0]
```
[the remark this answers is not in the document]
Comment (laser) [reply internal/app/go-filecoin/storage_market_connector/client.go]: fixed

File: internal/app/go-filecoin/storage_miner_connector/connector.go [hunk 3]
```diff
@@ -128,46 +130,46 @@ func (m *StorageMinerNodeConnector) handleNewTipSet(ctx context.Context, previou
...
-			ProviderCollateral:   big.Zero(),
-			ClientCollateral:     big.Zero(),
+		proposals[i] = market.ClientDealProposal{
+			Proposal: market.DealProposal{
+				PieceCID:             piece.PieceCID,
+				PieceSize:            piece.Size,
+				Client:               waddr,
+				Provider:             m.minerAddr,
+				StartEpoch:           0, // TODO: Does this have to be set to current height?
+				EndEpoch:             abi.ChainEpoch(math.MaxInt32),
```
[the remark this answers is not in the document]
Comment (laser) [reply internal/app/go-filecoin/storage_miner_connector/connector.go]: understood

File: internal/app/go-filecoin/storage_market_connector/client.go [hunk 4]
```diff
@@ -161,41 +160,41 @@ func (s *StorageClientNodeConnector) ValidatePublishedDeal(ctx context.Context,
...
 
 // SignProposal uses the local wallet to sign the given proposal
-func (s *StorageClientNodeConnector) SignProposal(ctx context.Context, signer address.Address, proposal *storagemarket.StorageDealProposal) error {
-	signFn := func(ctx context.Context, data []byte) (*smtypes.Signature, error) {
-		return s.SignBytes(ctx, signer, data)
+func (s *StorageClientNodeConnector) SignProposal(ctx context.Context, signer address.Address, proposal market.DealProposal) (*market.ClientDealProposal, error) {
+	buf := new(bytes.Buffer)
+	if err := proposal.MarshalCBOR(buf); err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (laser) [reply internal/app/go-filecoin/storage_market_connector/client.go]: fixed

File: internal/app/go-filecoin/storage_miner_connector/connector.go [hunk 5]
```diff
@@ -403,3 +422,87 @@ func (m *StorageMinerNodeConnector) waitForMessageHeight(ctx context.Context, mc
...
 	}
 }
+
+func (m *StorageMinerNodeConnector) GetMinerWorkerAddressFromChainHead(ctx context.Context, maddr address.Address) (address.Address, error) {
+	ts, err := m.chainStore.GetTipSet(m.chainStore.GetHead())
+	if err != nil {
+		return address.Undef, xerrors.Errorf("getting head ts for SealTicket failed: %w", err)
+	}
+
+	return m.workerGetter(ctx, maddr, ts.Key())
```
[the remark this answers is not in the document]
Comment (laser) [reply internal/app/go-filecoin/storage_miner_connector/connector.go]: I've updated to use the new state view structure, as per your request.

**Pull request merged**